azurelinuxagent/common/protocol/wire.py

```diff
@@ -825,39 +860,56 @@ def check_wire_protocol_version(self):
             raise ProtocolNotFoundError(error)
 
     def upload_status_blob(self):
-        ext_conf = self.get_ext_conf()
+        update_goal_state = False
+
+        while True:
```

Comment:
I see the reason for changing this to a while loop, but having it open ended makes me nervous. does it make sense to put in a maximum retry limit?